lib/path/default.nix

```
@@ -149,6 +173,51 @@ in /* No rec! Add dependencies on this file at the top. */ {
           ${subpathInvalidReason subpath}'';
     path + ("/" + subpath);
 
+  /*
+  Whether the first path is a prefix of the second path.
```

Maybe also something like

```suggestion
  Whether the first path is a component-wise prefix of the second path.
```

To make it clear that `hasPrefix /foo/b /foo/bar == false`.

---

Sounds good, applied!